### PR TITLE
Update to agent0 for pypechain update

### PR DESCRIPTION
**Pull request by slundqui**

### CODE REVIEW

File: scripts/fork_fuzz_bots.py [hunk 1]
```diff
@@ -66,43 +66,31 @@ def _fuzz_ignore_errors(exc: Exception) -> bool:
...
-        ):
+        if isinstance(orig_exception, ContractCustomError) and exc.decoded_error == "DistributeExcessIdleFailed()":
             return True
 
         # MinimumTransactionAmount error
-        if (
-            isinstance(orig_exception, ContractCustomError)
-            and "ContractCustomError('MinimumTransactionAmount')" in exc.args
-        ):
+        if isinstance(orig_exception, ContractCustomError) and exc.decoded_error == "MinimumTransactionAmount()":
```
Comment (dpaiton) [inline scripts/fork_fuzz_bots.py:87]: much better!

File: src/agent0/ethpy/hyperdrive/addresses.py [hunk 2]
```diff
@@ -78,7 +79,9 @@ def get_hyperdrive_addresses_from_registry(hyperdrive_registry_addr: str, web3:
     # Call registry contract to get registered pools.
     num_instances = registry_contract.functions.getNumberOfInstances().call()
     hyperdrive_addresses = registry_contract.functions.getInstancesInRange(0, num_instances).call()
-    hyperdrive_infos = registry_contract.functions.getInstanceInfosWithMetadata(hyperdrive_addresses).call()
+    # TODO pypechain needs a list input for any functions that takes a vector as an input.
```
Comment (dpaiton) [inline src/agent0/ethpy/hyperdrive/addresses.py:82]: https://github.com/delvtech/pypechain/issues/144

Comment (slundqui) [reply src/agent0/ethpy/hyperdrive/addresses.py:82]: 🙏

File: src/agent0/ethpy/hyperdrive/interface/_contract_calls.py [hunk 3]
```diff
@@ -93,9 +89,14 @@ def _get_vault_shares(
         # Type narrowing
         assert interface.morpho_contract is not None
         assert interface.morpho_market_id is not None
+
+        # TODO pypechain requires bytes input (not HexBytes) for the position function call.
```
Comment (dpaiton) [inline src/agent0/ethpy/hyperdrive/interface/_contract_calls.py:93]: https://github.com/delvtech/pypechain/issues/145

File: src/agent0/ethpy/hyperdrive/interface/_contract_calls.py [hunk 4]
```diff
@@ -321,52 +322,50 @@ async def _async_open_long(
...
-            FixedPoint(scaled_value=preview_result["bondProceeds"]) * (FixedPoint(1) - slippage_tolerance)
+            FixedPoint(scaled_value=preview_result.bondProceeds) * (FixedPoint(1) - slippage_tolerance)
         ).scaled_value
         fn_args = (
             trade_amount.scaled_value,
             min_output,
             min_vault_share_price,
-            (  # IHyperdrive.Options
-                agent_checksum_address,  # destination
+            Options(  # IHyperdrive.Options
```
Comment (dpaiton) [inline src/agent0/ethpy/hyperdrive/interface/_contract_calls.py:349]: 🎉 🎉

File: src/agent0/ethpy/hyperdrive/interface/_contract_calls.py [hunk 5]
```diff
@@ -410,50 +414,46 @@ async def _async_close_long(
...
     )
 
-    # To catch any solidity errors, we always preview transactions on the current block
-    # before calling smart contract transact
-    # Since current_pool_state.block_number is a property, we want to get the static block here
-    preview_result = {}
+    contract_fn = interface.hyperdrive_contract.functions.closeLong(*fn_args)
+    tx_params = TxParams({"from": sender.address})
+    preview_result = None
     if preview_before_trade or slippage_tolerance is not None:
```
Comment (dpaiton) [inline src/agent0/ethpy/hyperdrive/interface/_contract_calls.py:427]: Wouldn't we still want this comment?

```py
# To catch any solidity errors, we always preview transactions on the current block
# before calling smart contract transact
```

File: src/agent0/ethpy/hyperdrive/interface/read_write_interface.py [hunk 6]
```diff
@@ -194,7 +194,7 @@ async def async_open_long(
         """
         return await _async_open_long(
             interface=self,
-            agent=agent,
```
Comment (dpaiton) [inline src/agent0/ethpy/hyperdrive/interface/read_write_interface.py:197]: I'm thinking we should rename the arg here (`read_write_interface.async_open_long` & elsewhere) to `sender` as well.

Comment (dpaiton) [reply src/agent0/ethpy/hyperdrive/interface/read_write_interface.py:197]: The `_*` fns are meant to be at parity with the interface.

**Review by dpaiton — APPROVED** [review]

**Pull request merged**